src/rgw/store/sfs/types.cc

```diff
@@ -86,6 +86,28 @@ void Object::metadata_finish(SFStore *store) {
   db_versioned_objs.store_versioned_object(*db_versioned_object);
 }
 
+int Object::delete_object_version(SFStore *store) {
+  // remove object data
+  std::filesystem::remove(store->get_data_path() / get_storage_path());
+
+  // remove metadata
+  sqlite::SQLiteVersionedObjects db_versioned_objs(store->db_conn);
+  db_versioned_objs.remove_versioned_object(version_id);
+  return 0;
+}
+
+void Object::delete_object(SFStore *store) {
+  // delete empty folders going up to the parent path starting at the uuid path
+  auto current_path = store->get_data_path() / path.to_path();
+  while(std::filesystem::is_empty(current_path)) {
```

Comment:
good point

Reply:
the problem is if the thread storing an object can deschedule after an hypothetical `ensure directory` and if in that case the GC can access to that bucket (is that mutex-like protected?)

Reply:
yup, but isn't it nice to have the store fs clean after the GC did its job? :D 
Now seriously, I used that for my tests, but I think we should remove it completely.

I think we can implement the deletion of empty folders in a future policy.
This is going to be a mess because if we block the fs for long the writers are going to be waiting (more delays).

I'm still not sure how to address this, tbh. 

For now it only deletes the object's folder.